internal/controller/postgrescluster/pgadmin.go

```diff
@@ -307,6 +308,20 @@ func (r *Reconciler) reconcilePGAdminStatefulSet(
 	if err == nil {
 		pgadmin.Pod(cluster, configmap, &sts.Spec.Template.Spec, dataVolume)
 	}
+
+	// add nss_wrapper init container and add nss_wrapper env vars to the pgAdmin
+	// container
+	addNSSWrapper(
```

Comment:
Agreed, that should be refactored. Looking at our other similar functions, I think I'll just return if `setControllerReference` has an error and otherwise just return the error value from apply. That should accomplish the same goal while (hopefully) looking a little cleaner.